zerver/tornado/event_queue.py

```diff
@@ -295,6 +296,7 @@ def empty(self) -> bool:
     # See the comment on pop; that applies here as well
     def prune(self, through_id: int) -> None:
         while len(self.queue) != 0 and self.queue[0]['id'] <= through_id:
+            self.newest_pruned_id = max(self.newest_pruned_id, self.queue[0]['id'])
```

Comment:
Any reason this couldn’t go outside the loop as `self.newest_pruned_id = max(self.newest_pruned_id, through_id)`?

Reply:
Well, `through_id` might not be a real event ID, so we'd probably want to call the field something different if we did that.  Maybe `prune_threshhold`?

It's also not entirely clear to me that we actually need the `max` here, as another implementation path.

Reply:
Yeah, the `max` is unnecessary because `self.queue[0]['id'] <= through_id`, the caller just checked `through_id <= self.newest_pruned_id`, and `self.newest_pruned_id` is, by induction, nondecreasing through the loop.

An advantage to doing it your way (`self.newest_pruned_id = max(self.newest_pruned_id, self.queue[0]['id'])`) is that we should be able to assert `self.newest_pruned_id == through_id` at the end of the loop to catch another potential client misbehavior.